runner.js

```diff
@@ -16,14 +16,34 @@ const version = require('./version');
 const {timeoutPromise} = require('./promise_utils');
 const { shouldShowError } = require('./output');
 
+/**
+ * @typedef {(config: import('./config') => Promise<void> | void)} TeardownHook
+ */
+
+// Work around JSDoc limitation, which doesn't seem to support extending
+// interfaces on the fly.
+/**
+ * @typedef {Object} RawTaskConfig
+ * @property {TeardownHook[]} teardown_hooks
+ * @property {import('puppeteer').Page[]} _browser_pages
+ * @property {string} _testName
+ * @property {string} _taskName
+ */
+
+/**
+ * @typedef {import('./config').Config & RawTaskConfig} TaskConfig
+ */
+
 /**
  * @param {import('./config').Config} config
  * @param {Task} task
  * @private
  */
 async function run_task(config, task) {
+    /** @type {TaskConfig} */
     const task_config = {
         ...config,
+        teardown_hooks: [],
```

Comment:
This would make `config.teardown_hooks` the public API. Can we instead use a helper function. `registerTeardown(config, func)`? That would allow us to later change the implementation. Plus, a function is generally easier to document.

Reply:
You're right, much better to have a function for that! Went with a slightly different name `onTeardown` if that's ok. Let me know what you think.